crates/matrix-sdk/src/notification_settings/mod.rs

```diff
@@ -199,6 +200,31 @@ impl NotificationSettings {
             }
         };
 
+        let ruleset = self.rules.read().await.clone().ruleset;
+        for rule_id in rule_ids {
+            // Unstable polls push rule may be missing. Skip the update in case.
+            if ruleset.underride.iter().any(|r| r.rule_id == rule_id.to_string()) {
+                self.set_underride_push_rule_actions(rule_id, actions.clone()).await?
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Sets the push rule actions for a given underride push rule
```

Comment:
Could you explain briefly what an `underride` push rule is in the doc? A link to the spec or MSC which defines this would be nice as well.

Actually, this method doesn't seem to be this useful and it breaks the flow of the `set_default_notification_mode` method, which now doesn't make a request to the server, unless this method gets called as well.

Reply:
> Could you explain briefly what an underride push rule is in the doc

Sure, I'll add it shortly.

> Actually, this method doesn't seem to be this useful and it breaks the flow of the set_default_notification_mode method, which now doesn't make a request to the server, unless this method gets called as well.

This method encapsulate the set of actions of a generic underride rule.
Since I need it both for messages and polls push rules I extracted it as a separate function.
Maybe I can make it private for now since no one use it outside this crate.
Does it make sense?

Reply:
> Sure, I'll add it shortly.

Applied [here](https://github.com/matrix-org/matrix-rust-sdk/pull/2644/commits/5be510131aead71842c891105f64e76bcf60a9fe)